dbt/include/athena/macros/adapters/persist_docs.sql

```diff
@@ -1,10 +1,12 @@
 {% macro athena__persist_docs(relation, model, for_relation, for_columns) -%}
   {% set persist_relation_docs = for_relation and config.persist_relation_docs() and model.description %}
   {% set persist_column_docs = for_columns and config.persist_column_docs() and model.columns %}
-  {% if (persist_relation_docs or persist_column_docs) and relation.type != 'view' %}
+  {% set skip_archive = not is_incremental() %}
```

Comment:
What is the purpose of this skipArchive ?

Reply:
When you create new Athena table during dbt run you have three table versions just after creation:
1. Version 0 - after CREATE TABLE AS
2. Version 1 - after ALTER TABLE
3. Version 2 - after persist doc with boto3 update_table()
The idea here is to skip storing of Version 1 with skip_archive and have only two versions after single dbt run (Version 0, Version 2). Ideally we should have only one version (Version 2) but I don't want to touch this ALTER TABLE stuff here.
Example: https://user-images.githubusercontent.com/21337619/246951104-05401f69-2511-4e86-9cb5-bfdf8636ab3c.png

What about views?
When you create new Athena view you have two versions after single dbt run:
1. Version 0 - after CREATE OR REPLACE VIEW AS
2. Version 1 - after persisting docs
Let's just skip archiving of Version 0  and have only Version 1 after table creation. After second run only Version 3 will be added. One run - one version, good.
Example: https://user-images.githubusercontent.com/21337619/246951060-1917e813-602c-4229-9d28-ecf80f3c8fc1.png

Incremenal tables?
Creation of incremental table does not differ from ordinary table. But what about incremental run?
1. Version 0 - after CREATE TABLE AS
2. Version 1 - after ALTER TABLE (skip it)
3. Version 2 - after persisting doc with boto3 update_table() - initial run result
4. Version 3 - after persisting doc with **skip_archive=False** - next incremental run. Only in this case we don't want to skip version creation because we want to preserve Version 2 created after initial run. After incremental run we have +1 version.
Example: https://user-images.githubusercontent.com/21337619/246951487-cbef6226-70bc-4608-ac67-d440f0e7ad72.png

Conclusion: implemented trick with boto3 skip_archive option allows us to reduce number of redundant temporary versions of Glue Tables.